swap/src/protocol/alice/event_loop.rs

```diff
@@ -105,34 +106,10 @@ pub async fn new<LR>(
         tokio::select! {
             swarm_event = swarm.next_event() => {
                 match swarm_event {
-                    SwarmEvent::Behaviour(OutEvent::SpotPriceRequested { request, channel, peer }) => {
-                        let btc = request.btc;
-                        let xmr = match handle_spot_price_request(&mut latest_rate, btc, max_buy, monero_wallet.clone()).await {
-                            Ok(xmr) => xmr,
-                            Err(e) => {
-                                tracing::warn!(%peer, "Failed to produce spot price for {}: {:#}", btc, e);
-                                continue;
-                            }
-                        };
-
-                        match swarm.spot_price.send_response(channel, spot_price::Response { xmr }) {
-                            Ok(_) => {},
-                            Err(_) => {
-                                // if we can't respond, the peer probably just disconnected so it is not a huge deal, only log this on debug
-                                tracing::debug!(%peer, "Failed to respond with spot price");
-                                continue;
-                            }
-                        }
-
-                        let state0 = match State0::new(btc, xmr, env_config, bitcoin_wallet.as_ref(), &mut OsRng).await {
-                            Ok(state) => state,
-                            Err(e) => {
-                                tracing::warn!(%peer, "Failed to make State0 for execution setup: {:#}", e);
-                                continue;
-                            }
+                    SwarmEvent::Behaviour(OutEvent::SpotPriceRequested { request: spot_price::Request { btc }, channel, peer }) => {
+                        if let Err(e) = handle_spot_price_request(&mut latest_rate, &mut swarm, monero_wallet.as_ref(), bitcoin_wallet.as_ref(), env_config, channel, peer, btc, max_buy).await {
```

Comment:
Looks like `fmt` did not pick up the formatting in the `select!` - might be worth manually adding a breakline.

Reply:
Yeah, anything in the macro is never formatted. because it is opaque code to the formatter.